src/api.ts

```diff
@@ -1,5 +1,12 @@
 import type { CID } from 'multiformats/cid'
 
+export type Promisable<T> = T | PromiseLike<T>
+
+/**
+ * Literally any `Iterable` (async or regular).
+ */
+export type AnyIterable<T> = Iterable<T> | AsyncIterable<T>
```

Comment:
And we also usually call this `AwaitIterable` elsewhere so I'd suggest using same here for consistency.